server/src/main/java/org/opensearch/extensions/rest/RestSendToExtensionAction.java

```diff
@@ -97,8 +99,13 @@ public RestChannelConsumer prepareRequest(final RestRequest request, final NodeC
         }
         String message = "Forwarding the request " + method + " " + uri + " to " + discoveryExtension;
         logger.info(message);
-        // Hack to pass a final class in to fetch the response string
-        final StringBuilder responseBuilder = new StringBuilder();
+        // Initialize response. Values will be changed in the handler.
+        final RestExecuteOnExtensionResponse restExecuteOnExtensionResponse = new RestExecuteOnExtensionResponse(
+            RestStatus.ACCEPTED,
```

Comment:
Do we need to keep the initial status as `ACCEPTED`? Can we initialize with `NOT_FOUND` or something in the range of 4xx?

Reply:
Good question. I scanned the list and thought this was a reasonable "current state" but on looking at the docs it seems to imply a "final" async call, so it's probably not right.  Not sure if 404 is better than 500 as a default otherwise.

I think it's a moot point, though, as it's always overwritten. With the countdown latch it will only return from the two methods that count down, or from a timeout.  So we could set it `null` and it probably wouldn't matter.
 - On success it's overwritten by the response sent by the extension's handler.
 - On exception it's a 500.
 - On timeout waiting for the response it's a 408.

Leaning toward 500 at this point, will see what other reviewers have to say.

Reply:
I'm fine with 500.

Reply:
Switching to 500 which will save me having to overwrite it on the exception.

Reply:
Okay. So I remembered, recently I worked on a [bug](https://github.com/opensearch-project/OpenSearch/pull/4246) for OpenSearch which asked the API response to be changed from 500 to 4xx because it should be an exception rather server error. I'm still thinking around should we keep it 500 or something on 4xx line. Let's see what other reviewers have to say.

Reply:
@saratvemulapalli any thoughts here?

Reply:
500 sounds perfect, its designed for server errors.